src/sync/sync_config.hpp

```diff
@@ -100,6 +104,7 @@ struct SyncConfig {
     SyncSessionStopPolicy stop_policy;
     std::function<SyncBindSessionHandler> bind_session_handler;
     std::function<SyncSessionErrorHandler> error_handler;
+    std::shared_ptr<ChangesetTransformer> transformer;
```

Comment:
Removing that line does indeed work. But this means that if a `sync::Session` outlives the `SharedRealm` then it could be holding onto an invalid cooker. If we removed this line and set the cooker using the history object would that still work?